pages/contentful/slug-validation/index.vue

```diff
@@ -0,0 +1,202 @@
+            // slug is the same as title, enable automatic update when the title changes.
+            this.slugFromTitle = true;
+          }
+        }
+      });
+    },
+
+    methods: {
+      updateContentfulField() {
+        if (this.contentfulExtensionSdk) this.contentfulExtensionSdk.field.setValue(this.value);
+      },
+
+      // Convert using the speakingurl package, which is included in the contentful layout.
+      convertToSlug(text) {
+        return window.getSlug(text);
+      },
+
+      /**
+       * Handle change of slug value caused by either changing slug field
+       * or changing the title of the entry
+       */
+      handleValueChange(value) {
+        this.value = value;
+        this.getDebouncedDuplicateStatus(value);
+      },
+
+      /**
+       * Handle change of slug value caused by changing the slug of the entry.
+       * This will disable updates of the slug from the title for the  current form.
+       */
+      handleSlugChange(value) {
+        this.slugFromTitle = false;
+        this.handleValueChange(value);
+      },
+
+      /**
+       * Handle change of title value caused by changing the title of the entry.
+       * If the slug was changed independently, don't update based of title changes.
+       */
+      handleTitleChange(value) {
+        if (this.slugFromTitle) this.handleValueChange(this.convertToSlug(value || ''));
+      },
+
+      /**
+       * Handle validation status, setting the field to valid/invalid and
+       * removing the value in order to prevent publication if the slug is a duplicate.
+       */
+      async handleStatus() {
+        const status = await this.duplicateStatus;
+        if (status) {
+          this.slugField.removeValue();
+          this.slugField.setInvalid(true);
+          this.errorMessage =  'This slug has already been published in another entry';
+        } else {
+          this.slugField.setInvalid(false);
+          this.errorMessage = null;
+        }
+      },
+
+      // To prevent multiple API calls, or validation from a previous slug when typing fast.
+      getDebouncedDuplicateStatus(slug) {
+        if (this.timeout) clearTimeout(this.timeout);
+        this.timeout = setTimeout(() => {
+          this.debouncedDuplicateStatus = this.getDuplicates(slug);
+        }, 500);
+      },
+      /**
+       * Check if slug is already in use.
+       * Resolves to 'true' if there are entries of the given content type that have
+       * the same 'slug' value.
+       */
+      async getDuplicates(slug) {
+        if (!slug) {
+          return Promise.resolve(false);
+        }
+
+        const query = {};
+
+        query['fields.' + this.slugField.id] = slug;
+        query['sys.id[ne]'] = this.contentfulExtensionSdk.entry.getSys().id;
+        query['sys.publishedAt[exists]'] = true;
+
+        query['content_type'] = 'staticPage';
+        const hasStaticPageMatch = this.contentfulExtensionSdk.space.getEntries(query).then((result) => {
```

Comment:
I'm not sure about this as it would by default use whichever contentful API connection the portal on which the app is running would use.

Using the contentfulExtensionSdk to query guarantees that the contentful connection is that for the current page/the one for the org/space/environment in which the current object is.

Reply:
Quite right.